models/siegert_neuron.h

```diff
@@ -0,0 +1,327 @@
+/*
+ *  siegert_neuron.h
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef SIEGERT_NEURON_H
+#define SIEGERT_NEURON_H
+
+#include "config.h"
+
+#ifdef HAVE_GSL
+
+// C includes:
+#include <gsl/gsl_integration.h>
+
+// Includes from nestkernel:
+#include "archiving_node.h"
+#include "connection.h"
+#include "event.h"
+#include "nest_types.h"
+#include "node.h"
+#include "normal_randomdev.h"
+#include "poisson_randomdev.h"
+#include "ring_buffer.h"
+#include "recordables_map.h"
+#include "universal_data_logger.h"
+
+namespace nest
+{
+
+
+/* BeginDocumentation
+Name: siegert_neuron
+
+Description:
+
+ siegert_neuron is an implementation of a rate model with the
+ non-linearity given by the gain function of the
+ leaky-integrate-and-fire neuron with delta or exponentially decaying
+ synapses. The model can be used for a mean-field analysis of spiking
+ networks.
```

Comment:
Perhaps, a reference to a paper with the actual equations would be useful here?
For instance:
Fourcaud, Nicolas and Brunel, Nicolas, "Dynamics of the firing probability of noisy integrate-and-fire neurons", 14 (2002), pp. 2057--2110.